test/broadway_test.exs

```diff
@@ -682,6 +682,35 @@ defmodule BroadwayTest do
     end
   end
 
+  describe "ack_immediately" do
+    test "generate batches based on :batch_size" do
+      broadway_name = new_unique_name()
+      test_pid = self()
+
+      handle_message = fn message, _ ->
+        message = Message.ack_immediately(message)
+        send(test_pid, :manually_acked)
```

Comment:
Let's add one more test that calls this with a batch of messages. Let's also fail one message in said batch to make sure we are splitting them properly.

Reply:
Good call! Just pushed a commit that does this :)